scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/common/task/Task.java

```diff
@@ -778,4 +802,45 @@ public void setForkEnvironment(ForkEnvironment forkEnvironment) {
         this.forkEnvironment = forkEnvironment;
     }
 
+    /**
+     * Returns whether the task should be run in a forked JVM, depending on the global configuration and task property.
+     * The global configuration has a higher priority than the task property; And runasme true implies fork true.
+     *
+     * @return whether the task should be run in a forked JVM.
+     */
+    public boolean isForkingTask() {
+        if (PASchedulerProperties.TASK_RUNASME.getValueAsBoolean()) {
```

Comment:
this is part of the scheduler-api module which is not necessarily used server-side

Reply:
I just moved this part (i.e., calculating task fork and runAsMe based on its parameter and global config) into a separated class in `scheduler-server/.../util/TaskConfiguration.java`.

p.s. I didn't move it into `internalTask`, because some of these functions are called before the internalTask is initialized.